packages/react-native-reanimated/Common/cpp/reanimated/Tools/FeatureFlags.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include <string>
+#include <unordered_map>
+
+namespace reanimated {
+
+class StaticFeatureFlags {
+ public:
+#ifdef REANIMATED_FEATURE_FLAGS
+
+// Convert the value under x into a string
+#define XTOSTRING(x) #x
+// Evaluate the flag value; without this step, it would stringify the flag name
+// itself instead of the flag value
+#define TOSTRING(x) XTOSTRING(x)
+
+  static constexpr bool getFlag(const std::string_view &name) {
+    std::string nameStr = name.data();
+    std::string featureFlags = TOSTRING(REANIMATED_FEATURE_FLAGS);
+    if (featureFlags.find("[" + nameStr + ":") == std::string::npos) {
+      throw std::logic_error("Unable to recognize flag: " + nameStr);
+    }
+    return featureFlags.find("[" + nameStr + ":true]") != std::string::npos;
+  }
+
+#else
+
+  static constexpr bool getFlag(const char *) {
+    return false;
+  }
+
+#endif
+};
+
+class DynamicFeatureFlags {
+  static std::unordered_map<std::string, bool> flags_;
+
+ public:
+  static bool getFlag(const std::string &name);
+  static void setFlag(const std::string &name, bool value);
+};
```

Comment:
Just a suggestion, so feel free to ignore this, but this way seems to have two benefits:
* You do not increase cognitive effort of reader to control that default access specifier in c++ classes is id `private` and you explicitly mark that this private
* Usually, code is read from top to bottom, so having private members below public sounds reasonable
```suggestion
class DynamicFeatureFlags {
 public:
  static bool getFlag(const std::string &name);
  static void setFlag(const std::string &name, bool value);

 private:
   static std::unordered_map<std::string, bool> flags_;
};
```